src/hazelcore/semantics/CursorInfo.re

```diff
@@ -946,6 +1049,101 @@ let ana_cursor_found_exp =
     );
   };
 
+let rec find_uses_block =
+        (x: Var.t, block: UHExp.block, steps, index): uses_list => {
+  let UHExp.Block(lines, e) = block;
+  let (uses, shadow) = find_uses_lines(x, lines, steps @ [0], index, false);
+  shadow
+    ? uses
+    : uses @ find_uses_exp(x, e, steps @ [index + List.length(lines)]);
+}
+and find_uses_lines =
+    (x: Var.t, lines: UHExp.lines, steps, index, shadow): (uses_list, bool) =>
+  switch (lines, shadow) {
+  | ([], shadow) => ([], shadow)
+  | (_, true) => ([], true)
+  | ([line, ...lines], false) =>
+    let (line_uses, shadow) = find_uses_line(x, line, steps @ [index]);
+    let (lines_uses, shadow) =
+      find_uses_lines(x, lines, steps, index + 1, shadow);
+    (line_uses @ lines_uses, shadow);
+  }
+and find_uses_line = (x: Var.t, line: UHExp.line, steps): (uses_list, bool) =>
+  switch (line) {
+  | ExpLine(e) => (find_uses_exp(x, e, steps), false)
+  | EmptyLine => ([], false)
+  | LetLine(p, _, block) => (
+      find_uses_block(x, block, steps @ [1], 0),
+      shadow_in_pat(x, p),
+    )
+  }
+and find_uses_exp = (x: Var.t, e: UHExp.t, steps): uses_list =>
+  switch (e) {
+  | EmptyHole(_)
+  | Var(_, InVarHole(_, _), _)
+  | Var(InHole(_, _), _, _)
```

Comment:
A variable in an ErrStatus hole (i.e. `InHole`) is still a use (it just happens to be a use in a type-incorrect position). Only when its `InVarHole` is it necessarily not a use of anything (because that hole means that it is unbound)